src/cascadia/TerminalSettingsModel/Profile.cpp

```diff
@@ -336,3 +338,159 @@ Json::Value Profile::ToJson() const
 
     return json;
 }
+winrt::hstring Profile::EvaluatedIcon()
+{
+    // We cache the result here, so we don't search the path for the exe every time.
+    if (!_evaluatedIcon.has_value())
+    {
+        _evaluatedIcon = _evaluateIcon();
+    }
+    return *_evaluatedIcon;
+}
+
+winrt::hstring Profile::_evaluateIcon() const
+{
+    // If the profile has an icon, return it.
+    if (!Icon().empty())
+    {
+        return Icon();
+    }
+
+    // Otherwise, use NormalizeCommandLine to find the actual exe name. This
+    // will actually search for the exe, including spaces, in the same way that
+    // CreateProcess does.
+    std::wstring cmdline{ NormalizeCommandLine(Commandline().c_str()) };
+    if (cmdline.empty())
+    {
+        return {};
+    }

```

Comment:
FWIW you can remove this check. An empty `wstring` still returns a valid string via `.c_str()` (`""`) and so all the code below should still work.